src/com/cft/contactmerge/contact/Address.java

```diff
@@ -77,6 +79,40 @@ public void setZip(String zip) {
      *************************************************** Sub-Methods ***************************************************
      *******************************************************************************************************************/
     public String toString() {
-        return getStreetAddress() + ", " + getApartment() + ", " + getCity() + ", " +
-                getState() + ", " + getCountry() + ", " + getZip(); }
+        StringBuilder fullAddress = new StringBuilder();
+        String[] addressParts = {getStreetAddress(), getApartment(), getCity(), getState(), getCountry(), getZip()};
```

Comment:
I just had to write something like this myself and found a nice version on StackOverflow… It avoids having the go back and remove the extra delimiter.     

// from https://stackoverflow.com/questions/4930939/preferred-idiom-for-joining-a-collection-of-strings-in-java
    private String joinCollectionOfStrings(Collection<String> collection, String delimeter)
    {
        Iterator<String> iterator = collection.iterator();
        StringBuilder sb = new StringBuilder();

        if (iterator.hasNext()) {
            sb.append(iterator.next());

            while (iterator.hasNext()) {
                sb.append(delimeter).append(iterator.next());
            }
        }

        return sb.toString();
    }

Reply:
Nice, I'll use this in the next pull.